actix-service/src/fn_service.rs

```diff
@@ -92,13 +94,13 @@ where
 ///     Ok(())
 /// }
 /// ```
-pub fn fn_factory_with_config<F, Fut, Cfg, Srv, Err>(
+pub fn fn_factory_with_config<F, Fut, Cfg, Srv, Req, Err>(
     f: F,
-) -> FnServiceConfig<F, Fut, Cfg, Srv, Err>
+) -> FnServiceConfig<F, Fut, Cfg, Srv, Req, Err>
```

Comment:
I believe `FnServiceConfig<F, Fut, Cfg, Req>` would be enough type. Srv and Err type are actuall associate types for Fut so they can be unconstraint.